### PR TITLE
Winscard improvements

### PR DESCRIPTION
**Pull request by TheBestTvarynka**
Hi,
In this PR, I improved the `crates/winscard` crate. I didn't change the emulated smart card implementation a lot. Basically, I just added new methods to the `WinScard` and `WinScardContext` traits and implemented them for the emulated smart card.
Some of these changes may seem unnecessary but I need them to provide better system-provided smart cards support. Feel free to ask any questions regarding these changes.

_Note. I think I'll be better to squash and merge this PR._

### CODE REVIEW

**Comment by TheBestTvarynka** [comment]
@CBenoit  The CI is happy now and the PR is ready for review.

File: ffi/Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ whoami = "1.4"
 sspi = { path = "..", features = ["network_client", "dns_resolver"] }
 ffi-types = { path = "../crates/ffi-types", features = ["winscard"], optional = true }
 picky-asn1-der = "0.4"
-
+uuid = { version = "1.4.1", default-features = false }
```
Comment (CBenoit) [inline ffi/Cargo.toml:28]: nitpick: only specify the minor number
```suggestion
uuid = { version = "1.4", default-features = false }
```

File: crates/winscard/Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,8 @@ rand_core = "0.6.4"
 sha1 = { version = "0.10.6", default-features = false }
 base64 = { version = "0.21" , optional = true }
 picky-asn1-der = { version = "0.4", optional = true }
+num-derive = "0.4.2"
+num-traits = { version = "0.2.18", default-features = false }
```
Comment (CBenoit) [inline crates/winscard/Cargo.toml:36]: nitpick: Only specify the minor number.
```suggestion
num-derive = "0.4"
num-traits = { version = "0.2", default-features = false }
```

File: crates/winscard/src/winscard.rs [hunk 3]
```diff
@@ -250,6 +413,118 @@ pub struct TransmitOutData {
     pub receive_pci: Option<IoRequest>,
 }
 
+/// This structure represents the result of the `SCardConnect` function.
+pub struct ScardConnectData {
+    /// Established smart card handle.
+    pub scard: Box<dyn WinScard>,
```
Comment (CBenoit) [inline crates/winscard/src/winscard.rs:419]: naming: Should it be named `handle`? I see things like `scard.scard` at some places.

```suggestion
    pub handle: Box<dyn WinScard>,
```

**Review by CBenoit — COMMENTED** [review]
Thank you! This is looking good to me.

> Note. I think I'll be better to squash and merge this PR.

Yes, feel free to squash it once you are satisfied!

**Comment by CBenoit** [comment]
> Note. I think I'll be better to squash and merge this PR.

Yes, feel free to squash it once you are satisfied!

**Review by CBenoit — APPROVED** [review]
Feel free to merge yourself since this is targeting a dev branch.

File: crates/winscard/src/winscard.rs (hunk 3 above)
Comment (TheBestTvarynka) [reply crates/winscard/src/winscard.rs]: agreed. done

File: crates/winscard/Cargo.toml (hunk 2 above)
Comment (TheBestTvarynka) [reply crates/winscard/Cargo.toml]: done

File: ffi/Cargo.toml (hunk 1 above)
Comment (TheBestTvarynka) [reply ffi/Cargo.toml]: done

**Pull request merged**